File: Java/Builder/Director.java

public class Director 
{
	private Builder builder;
	
	public Director(Builder builder) {
		this.builder = builder;
	}
	
	public void construct()
	{
		this.builder.buildDefinition();
		this.builder.buildNumber();
		this.builder.buildValue();
	}
	
	public ComplexObject getProduct()
	{
		return this.builder.getObject();
	}
}

File: Swift/Patterns/Bridge/Bridge/Director.swift
//
//  Director.swift
//  Bridge
//
//  Created by colan biemer on 4/4/15.
//  Copyright (c) 2015 colan biemer. All rights reserved.
//

import Foundation

class Director
{
	var builder: Builder
	init(bld: Builder)
	{
		builder = bld
	}
	
	func construct()
	{
		builder.buildDefinition()
		builder.buildNumber()
		builder.buildValue()
	}
	
	func getObj() -> ComplexObject
	{
		return builder.getObj()
	}
}